src/OutputFormatter/JsonOutputFormatter.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\OutputFormatter;
+
+use function json_encode;
+use Qossmic\Deptrac\Console\Command\AnalyzeCommand;
+use Qossmic\Deptrac\Console\Output;
+use Qossmic\Deptrac\RulesetEngine\Context;
+use Qossmic\Deptrac\RulesetEngine\SkippedViolation;
+use Qossmic\Deptrac\RulesetEngine\Uncovered;
+use Qossmic\Deptrac\RulesetEngine\Violation;
+
+final class JsonOutputFormatter implements OutputFormatterInterface
+{
+    public const DUMP_JSON = 'json-dump';
+
+    public function getName(): string
+    {
+        return 'json';
+    }
+
+    /**
+     * @return OutputFormatterOption[]
+     */
+    public function configureOptions(): array
+    {
+        return [
+            OutputFormatterOption::newValueOption(self::DUMP_JSON, 'path to a dumped json file'),
+        ];
+    }
+
+    public function enabledByDefault(): bool
+    {
+        return false;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \Exception
+     */
+    public function finish(
+        Context $context,
+        Output $output,
+        OutputFormatterInput $outputFormatterInput
+    ): void {
+        $reportSkipped = $outputFormatterInput->getOptionAsBoolean(AnalyzeCommand::OPTION_REPORT_SKIPPED);
+        $reportUncovered = $outputFormatterInput->getOptionAsBoolean(AnalyzeCommand::OPTION_REPORT_UNCOVERED);
+
+        $jsonArray = [];
+        $violations = [];
+        foreach ($context->rules() as $rule) {
+            if (!$rule instanceof Violation && !$rule instanceof SkippedViolation && !$rule instanceof Uncovered) {
+                continue;
+            }
+
+            if (!$reportSkipped && $rule instanceof SkippedViolation) {
+                continue;
+            }
+
+            if (!$reportUncovered && $rule instanceof Uncovered) {
+                continue;
+            }
+
+            switch (true) {
+                case $rule instanceof Violation:
+                    $this->addFailure($violations, $rule);
+                    break;
+                case $rule instanceof SkippedViolation:
+                    $this->addSkipped($violations, $rule);
+                    break;
+                case $rule instanceof Uncovered:
+                    $this->addUncovered($violations, $rule);
+                    break;
+            }
+        }
+
+        $jsonArray['Report'] = [
+            'Violations' => count($context->violations()),
+            'Skipped violations' => count($context->skippedViolations()),
+            'Uncovered' => count($context->uncovered()),
+            'Allowed' => count($context->allowed()),
+            'Warnings' => count($context->warnings()),
+            'Errors' => count($context->errors()),
+        ];
+
+        foreach ($violations as &$value) {
+            $value['violations'] = count($value['messages']);
+        }
+
+        $jsonArray['files'] = $violations;
+        $json = json_encode($jsonArray);
```

Comment:
```suggestion
        $json = json_encode($jsonArray, \JSON_PRETTY_PRINT);
```
Please add the flag for `JSON_PRETTY_PRINT` to have nicer output when the json is dumped to the console.
Also, please add error handling in case `json_encode()` returns false.